### DISCUSSION TITLE
ERROR: Cannot compile a minimal program. The toolchain or QMakeSpec is broken.

### DISCUSSION THREAD

**Discussion by mtsnuc** [abhiTronix/raspberry-pi-cross-compilers#88] [General]
Hi,
first of all thank you for your tutorial to crosscompile QT for rasp and for your toolchains.
Anyway, I'm getting an error on QT configure:
```

Checking for valid makespec... 
+ cd /opt/qt5pi2021/build/config.tests/verifyspec && /opt/qt5pi2021/build/qtbase/bin/qmake "CONFIG -= qt debug_and_release app_bundle lib_bundle" "CONFIG += shared warn_off console single_arch" 'QMAKE_LIBDIR += /opt/qt5pi2021/sysroot/usr/lib/arm-linux-gnueabihf' 'INCLUDEPATH += /opt/qt5pi2021/sysroot/usr/include/arm-linux-gnueabihf' "QMAKE_CFLAGS += --sysroot=/opt/qt5pi2021/sysroot/" "QMAKE_CXXFLAGS += --sysroot=/opt/qt5pi2021/sysroot/" "QMAKE_LFLAGS += --sysroot=/opt/qt5pi2021/sysroot/" -early "CONFIG += cross_compile" /opt/qt5pi2021/qt-everywhere-src-5.15.2/qtbase/config.tests/verifyspec
+ cd /opt/qt5pi2021/build/config.tests/verifyspec && MAKEFLAGS= /usr/bin/make
> /opt/qt5pi2021/tools/cross-pi-gcc-10.2.0-2/bin/arm-linux-gnueabihf-g++ -c -march=armv8-a -mtune=cortex-a72 -mfpu=crypto-neon-fp-armv8 -mfloat-abi=hard --sysroot=/opt/qt5pi2021/sysroot/ -O2 -w -fPIC  -I/opt/qt5pi2021/qt-everywhere-src-5.15.2/qtbase/config.tests/verifyspec -I. -I/opt/qt5pi2021/sysroot/usr/include/arm-linux-gnueabihf -I/opt/qt5pi2021/qt-everywhere-src-5.15.2/qtbase/mkspecs/devices/linux-rasp-pi4-v3d-g++ -o verifyspec.o /opt/qt5pi2021/qt-everywhere-src-5.15.2/qtbase/config.tests/verifyspec/verifyspec.cpp
> /opt/qt5pi2021/tools/cross-pi-gcc-10.2.0-2/bin/arm-linux-gnueabihf-g++ -mfloat-abi=hard --sysroot=/opt/qt5pi2021/sysroot/ -Wl,-O1 -Wl,-rpath-link,/opt/qt5pi2021/sysroot/usr/lib/arm-linux-gnueabihf -Wl,-rpath-link,/opt/qt5pi2021/sysroot/lib/arm-linux-gnueabihf -o verifyspec verifyspec.o   -L/opt/qt5pi2021/sysroot/usr/lib/arm-linux-gnueabihf   
> /opt/qt5pi2021/tools/cross-pi-gcc-10.2.0-2/bin/../lib/gcc/arm-linux-gnueabihf/10.2.0/../../../../arm-linux-gnueabihf/bin/ld: -lstdc++ kann nicht gefunden werden
> collect2: Fehler: ld gab 1 als Ende-Status zurück
> make: *** [Makefile:68: verifyspec] Fehler 1
Note: Also available for Linux: linux-clang linux-icc

ERROR: Cannot compile a minimal program. The toolchain or QMakeSpec is broken.

```
What could this be? I followed your guide step by step. With another toolchain I was able configure, but I had problems with EGLFS.

I'm using a rasp4 with linux bullseye.

**Comment by abhiTronix** [comment]
@mtsnuc Toolchains are probably broken for Bullseye. I'm looking into it.

**Comment by mtsnuc** [comment]
That would be great!
Based on this table [Cross Compilers](https://github.com/abhiTronix/raspberry-pi-cross-compilers) I thought bullseye is already working.
I was able to run "configure" with some toolchains from other websites but they always have the problem of not detecting EGLFs correctly (GLFS Raspberry Pi ................... yes - even though I'm using a PI 4).
I hope you can fix it, so there is a working toolchain for PI4 on bullseye.

Thank you in advance!

**Comment by abhiTronix** [comment]
@mtsnuc This issue has been resolved in the latest Cross Compiler Toolchains version `v3.0.9`. Download it from [here](https://sourceforge.net/projects/raspberry-pi-cross-compilers/files/Raspberry%20Pi%20GCC%20Cross-Compiler%20Toolchains/Bullseye/GCC%2010.2.0/Raspberry%20Pi%201%2C%20Zero/) and Rerun QT instructions again. Goodluck!

**Comment by mtsnuc** [comment]
Hi. THANKS!
It runs without any error now, but why does it show "EGLFS Raspberry Pi ................... yes"
This should be no, otherwise it will not use MESA.

```
QPA backends:
  DirectFB ............................... no
  EGLFS .................................. yes
  EGLFS details:
    EGLFS OpenWFD ........................ no
    EGLFS i.Mx6 .......................... no
    EGLFS i.Mx6 Wayland .................. no
    EGLFS RCAR ........................... no
    EGLFS EGLDevice ...................... yes
    EGLFS GBM ............................ yes
    EGLFS VSP2 ........................... no
    EGLFS Mali ........................... no
    EGLFS Raspberry Pi ................... yes
    EGLFS X11 ............................ yes
```

**Reply by abhiTronix** [reply]
Have you done this step: https://github.com/abhiTronix/raspberry-pi-cross-compilers/blob/master/QT_build_instructions.md#4-enable-gl-fake-kms ?

**Reply by abhiTronix** [reply]
Try step 14 on https://wiki.qt.io/RaspberryPi2EGLFS and then rysnc again.

**Reply by mtsnuc** [reply]
Fake KMS is no longer used in bullseye.

**Reply by abhiTronix** [reply]
> Fake KMS is no longer used in bullseye.

Really, I didn't know that. I'll look into it.

**Reply by abhiTronix** [reply]
@mtsnuc No you can enable fake KMS by putting `dtoverlay=vc4-fkms-v3d` into your `/boot/config.txt`. Then, Restart to apply it.

**Reply by mtsnuc** [reply]
sure, but as KMS is now fully supported (and fake KMS is not needed anymore) I think it would make more sense to use it.

**Reply by abhiTronix** [reply]
Then see this issue: https://github.com/abhiTronix/raspberry-pi-cross-compilers/issues/77
